pkg/cloud/gcp/actuators/machineset/controller.go

```diff
@@ -156,6 +157,19 @@ func (r *Reconciler) reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, e
 	machineSet.Annotations[cpuKey] = strconv.FormatInt(machineType.GuestCpus, 10)
 	machineSet.Annotations[memoryKey] = strconv.FormatInt(machineType.MemoryMb, 10)
 
+	if (len(providerConfig.GuestAccelerators)) == 0 {
+		if len(machineType.Accelerators) == 0 {
+			machineSet.Annotations[gpuKey] = strconv.FormatInt(0, 10)
+		} else {
+			// Accelerators will always be max size of 1
+			machineSet.Annotations[gpuKey] = strconv.FormatInt(machineType.Accelerators[0].GuestAcceleratorCount, 10)
+		}
+
+	} else {
+		// Guest accelerators will always be max size of 1
+		machineSet.Annotations[gpuKey] = strconv.FormatInt(providerConfig.GuestAccelerators[0].AcceleratorCount, 10)
+	}
```

Comment:
This seems quite hard to follow right now with these else statements, wonder if this might be easier to understand as a switch. Switches are ordered so the top case will take priority. WDYT?

```
switch {
case len(providerConfig.GuestAccelerators) > 0:
  machineSet.Annotations[gpuKey] = strconv.FormatInt(providerConfig.GuestAccelerators[0].AcceleratorCount, 10)
case len(machineType.Accelerators) > 0:
  machineSet.Annotations[gpuKey] = strconv.FormatInt(machineType.Accelerators[0].GuestAcceleratorCount, 10)
default:
  machineSet.Annotations[gpuKey] = strconv.FormatInt(0, 10)
}
```